// @buf/digibear_digibear.bufbuild_es@1.3.3-20231008085331-f131bfe24b83.1/usr/v1/usr_service_pb.js
// @generated by protoc-gen-es v1.3.3
// @generated from file usr/v1/usr_service.proto (package usr.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck

import { proto3 } from "@bufbuild/protobuf";
import { Profile } from "./profile_pb.js";
import { AccountForProfileSelection } from "./account_for_profile_selection_pb.js";

/**
 * ::::: CREATE ::::: //
 * ::::: CREATE ::::: //
 * ::::: CREATE ::::: //
 * CreateProfile
 *
 * @generated from message usr.v1.CreateProfileRequest
 */
export const CreateProfileRequest = proto3.makeMessageType(
  "usr.v1.CreateProfileRequest",
  () => [
    { no: 1, name: "profile", kind: "message", T: Profile },
  ],
);

/**
 * @generated from message usr.v1.CreateProfileResponse
 */
export const CreateProfileResponse = proto3.makeMessageType(
  "usr.v1.CreateProfileResponse",
  () => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);

/**
 * ::::: READ ::::: //
 * ::::: READ ::::: //
 * ::::: READ ::::: //
 * GetProfile
 *
 * @generated from message usr.v1.GetProfileRequest
 */
export const GetProfileRequest = proto3.makeMessageType(
  "usr.v1.GetProfileRequest",
  () => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);

/**
 * @generated from message usr.v1.GetProfileResponse
 */
export const GetProfileResponse = proto3.makeMessageType(
  "usr.v1.GetProfileResponse",
  () => [
    { no: 1, name: "profile", kind: "message", T: Profile },
  ],
);

/**
 * CheckEmailInUse
 *
 * @generated from message usr.v1.CheckEmailInUseRequest
 */
export const CheckEmailInUseRequest = proto3.makeMessageType(
  "usr.v1.CheckEmailInUseRequest",
  () => [
    { no: 1, name: "email", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);

/**
 * @generated from message usr.v1.CheckEmailInUseResponse
 */
export const CheckEmailInUseResponse = proto3.makeMessageType(
  "usr.v1.CheckEmailInUseResponse",
  () => [
    { no: 1, name: "isEmailInUse", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
  ],
);

/**
 * GetAccountForProfileSelection
 *
 * @generated from message usr.v1.GetAccountForProfileSelectionRequest
 */
export const GetAccountForProfileSelectionRequest = proto3.makeMessageType(
  "usr.v1.GetAccountForProfileSelectionRequest",
  () => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);

/**
 * @generated from message usr.v1.GetAccountForProfileSelectionResponse
 */
export const GetAccountForProfileSelectionResponse = proto3.makeMessageType(
  "usr.v1.GetAccountForProfileSelectionResponse",
  () => [
    { no: 1, name: "accountForProfileSelection", kind: "message", T: AccountForProfileSelection },
  ],
);

/**
 * ::::: UPDATE ::::: //
 * ::::: UPDATE ::::: //
 * ::::: UPDATE ::::: //
 * UpdateProfile
 *
 * @generated from message usr.v1.UpdateProfileRequest
 */
export const UpdateProfileRequest = proto3.makeMessageType(
  "usr.v1.UpdateProfileRequest",
  () => [
    { no: 1, name: "profile", kind: "message", T: Profile },
  ],
);

/**
 * @generated from message usr.v1.UpdateProfileResponse
 */
export const UpdateProfileResponse = proto3.makeMessageType(
  "usr.v1.UpdateProfileResponse",
  () => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);

/**
 * ::::: DELETE ::::: //
 * ::::: DELETE ::::: //
 * ::::: DELETE ::::: //
 * DeleteProfile
 *
 * @generated from message usr.v1.DeleteProfileRequest
 */
export const DeleteProfileRequest = proto3.makeMessageType(
  "usr.v1.DeleteProfileRequest",
  () => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);

/**
 * @generated from message usr.v1.DeleteProfileResponse
 */
export const DeleteProfileResponse = proto3.makeMessageType(
  "usr.v1.DeleteProfileResponse",
  () => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);
