// @buf/digibear_digibear.connectrpc_es@1.1.2-20231008085331-f131bfe24b83.1/usr/v1/usr_service_connect.js
// @generated by protoc-gen-connect-es v1.1.2
// @generated from file usr/v1/usr_service.proto (package usr.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck

import { CheckEmailInUseRequest, CheckEmailInUseResponse, CreateProfileRequest, CreateProfileResponse, DeleteProfileRequest, DeleteProfileResponse, GetAccountForProfileSelectionRequest, GetAccountForProfileSelectionResponse, GetProfileRequest, GetProfileResponse, UpdateProfileRequest, UpdateProfileResponse } from "@buf/digibear_digibear.bufbuild_es/usr/v1/usr_service_pb.js";
import { MethodKind } from "@bufbuild/protobuf";

/**
 * @generated from service usr.v1.UsrService
 */
export const UsrService = {
  typeName: "usr.v1.UsrService",
  methods: {
    /**
     * CREATE
     *
     * @generated from rpc usr.v1.UsrService.CreateProfile
     */
    createProfile: {
      name: "CreateProfile",
      I: CreateProfileRequest,
      O: CreateProfileResponse,
      kind: MethodKind.Unary,
    },
    /**
     * READ
     *
     * @generated from rpc usr.v1.UsrService.GetProfile
     */
    getProfile: {
      name: "GetProfile",
      I: GetProfileRequest,
      O: GetProfileResponse,
      kind: MethodKind.Unary,
    },
    /**
     * @generated from rpc usr.v1.UsrService.CheckEmailInUse
     */
    checkEmailInUse: {
      name: "CheckEmailInUse",
      I: CheckEmailInUseRequest,
      O: CheckEmailInUseResponse,
      kind: MethodKind.Unary,
    },
    /**
     * @generated from rpc usr.v1.UsrService.GetAccountForProfileSelection
     */
    getAccountForProfileSelection: {
      name: "GetAccountForProfileSelection",
      I: GetAccountForProfileSelectionRequest,
      O: GetAccountForProfileSelectionResponse,
      kind: MethodKind.Unary,
    },
    /**
     * UPDATE
     *
     * @generated from rpc usr.v1.UsrService.UpdateProfile
     */
    updateProfile: {
      name: "UpdateProfile",
      I: UpdateProfileRequest,
      O: UpdateProfileResponse,
      kind: MethodKind.Unary,
    },
    /**
     * DELETE
     *
     * @generated from rpc usr.v1.UsrService.DeleteProfile
     */
    deleteProfile: {
      name: "DeleteProfile",
      I: DeleteProfileRequest,
      O: DeleteProfileResponse,
      kind: MethodKind.Unary,
    },
  }
};
